patches/minecraft/net/minecraft/world/biome/BiomeProvider.java.patch

```diff
@@ -14,21 +14,30 @@
      }
  
      private BiomeProvider(long p_i45744_1_, WorldType p_i45744_3_, String p_i45744_4_)
-@@ -39,6 +40,7 @@
++    { this(p_i45744_1_, p_i45744_3_, p_i45744_4_, null); }
++    public BiomeProvider(WorldInfo p_i46712_1_, net.minecraft.world.World world) // Forge: Added a World argument
++    { this(p_i46712_1_.func_76063_b(), p_i46712_1_.func_76067_t(), p_i46712_1_.func_82571_y(), world); }
++    private BiomeProvider(long p_i45744_1_, WorldType p_i45744_3_, String p_i45744_4_, net.minecraft.world.World world) // Forge: Added a World argument
+     {
+         this();
+ 
+@@ -38,7 +43,8 @@
+             this.field_190945_a = ChunkGeneratorSettings.Factory.func_177865_a(p_i45744_4_).func_177864_b();
          }
  
-         GenLayer[] agenlayer = GenLayer.func_180781_a(p_i45744_1_, p_i45744_3_, this.field_190945_a);
-+        agenlayer = getModdedBiomeGenerators(p_i45744_3_, p_i45744_1_, agenlayer);
+-        GenLayer[] agenlayer = GenLayer.func_180781_a(p_i45744_1_, p_i45744_3_, this.field_190945_a);
++        GenLayer[] agenlayer = GenLayer.initializeAllBiomeGenerators(p_i45744_1_, p_i45744_3_, this.field_190945_a, world);
++        agenlayer = getModdedBiomeGenerators(null, p_i45744_3_, p_i45744_1_, agenlayer);
          this.field_76944_d = agenlayer[0];
          this.field_76945_e = agenlayer[1];
      }
-@@ -207,6 +209,13 @@
+@@ -207,6 +213,13 @@
          this.field_76942_f.func_76838_a();
      }
  
-+    public GenLayer[] getModdedBiomeGenerators(WorldType worldType, long seed, GenLayer[] original)
++    public GenLayer[] getModdedBiomeGenerators(@javax.annotation.Nullable net.minecraft.world.World world, WorldType worldType, long seed, GenLayer[] original)
```

Comment:
Public methods like this can't be changed directly, they need to be deprecated and replaced with new methods.